test/test_blocks_on_demand.py

```diff
@@ -0,0 +1,303 @@
+"""
+Test blocks on demand mode.
+"""
+
+import pytest
+import requests
+from starkware.starknet.definitions.error_codes import StarknetErrorCode
+
+from .account import get_estimated_fee, invoke
+from .settings import APP_URL
+from .shared import (
+    ABI_PATH,
+    CONTRACT_PATH,
+    PREDEPLOY_ACCOUNT_CLI_ARGS,
+    PREDEPLOYED_ACCOUNT_ADDRESS,
+    PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+)
+from .test_state_update import get_state_update
+from .test_transaction_trace import get_block_traces
+from .util import (
+    ReturnCodeAssertionError,
+    assert_equal,
+    assert_hex_equal,
+    assert_tx_status,
+    call,
+    deploy,
+    devnet_in_background,
+    get_block,
+)
+
```

Comment:
The tests look solid but I need to read them again a couple of times.

Also, I'm not sure if we should test events in these test cases, they are missing here but I don't know if we should test it here or not. I need to meditate on that today.

Reply:
maybe try something with this:

```
@pytest.mark.usefixtures("run_devnet_in_background")
@pytest.mark.parametrize(
    "run_devnet_in_background, input_data, expected_data",
    GET_EVENTS_TEST_DATA,
    indirect=True,
)
def test_get_events(input_data, expected_data):
```

Reply:
Addressed in https://github.com/Shard-Labs/starknet-devnet/pull/389/commits/7f784e39dc46e3a2042dde202b3195d1724003d1